hugo/static/cgi-bin/watermark.cgi

```diff
@@ -0,0 +1,166 @@
+#!/opt/venv/watermark/bin/python3
+# -*- coding: utf-8 -*-
+"""
+watermark.cgi - On-the-fly PDF footer + page number service for the ACL Anthology.
+
+POST a multipart/form-data request with fields:
+  pdf          (file, required)  The input PDF
+  footer_text  (text, optional)  First-page footer block; use <i>…</i> for italics; newlines allowed
+  page_start   (int, optional)   Starting page number (>=1)
+
+Returns: Modified PDF (application/pdf) with Content-Disposition: attachment.
+
+Security / Resource considerations:
+  * Rejects files > 30MB (quick limit; adjust as needed).
+  * Basic PDF validation (magic header %PDF-).
+  * Uses temporary files; they are deleted at end of request.
+  * No persistent storage.
+
+Depends on: reportlab, pypdf (already required by bin/add_footer.py)
+"""
+
+import cgi
+import os
+import sys
+import tempfile
+import subprocess
+from pathlib import Path
+import shutil
+import urllib.parse
+MAX_BYTES = 30 * 1024 * 1024
+
+def http_error(status_code: int, message: str):
+    print(f"Status: {status_code} Bad Request" if status_code == 400 else f"Status: {status_code}")
+    print("Content-Type: text/plain; charset=utf-8")
+    print("X-Content-Type-Options: nosniff")
+    print()
+    print(message)
+    sys.exit(0)
+
+def find_add_footer() -> Path:
+    """Resolve location of add_footer.py relative to this CGI script."""
+    here = Path(__file__).resolve()
+    homedir = Path(os.environ.get('HOME', '/home/anthologizer'))
+    # Walk up to locate bin/add_footer.py
+    for parent in [here.parent, *here.parents, homedir / "acl-anthology"]:
+        candidate = parent.parent / "bin" / "add_footer.py" if parent.name == 'cgi-bin' else parent / "bin" / "add_footer.py"
+        if candidate.exists():
+            return candidate
+    # Fallback relative guess (3 levels up)
+    guess = here.parents[2] / 'bin' / 'add_footer.py'
+    return guess
+
+def main():
+    # Debug flag: append ?debug=1 to request URL to get full stderr on failure.
+    qs = os.environ.get('QUERY_STRING', '')
+    qparams = dict(urllib.parse.parse_qsl(qs, keep_blank_values=True)) if qs else {}
+    debug_mode = qparams.get('debug') in {'1','true','yes','on'}
+    method = os.environ.get('REQUEST_METHOD', 'GET').upper()
+    if method != 'POST':
+        http_error(405, 'Use POST with multipart/form-data.')
+
+    try:
+        length = int(os.environ.get('CONTENT_LENGTH', '0'))
+    except ValueError:
+        length = 0
+    if length <= 0:
+        http_error(400, 'Empty request body.')
+    if length > MAX_BYTES:
+        http_error(400, f'File too large (> {MAX_BYTES//1024//1024}MB).')
+
+    form = cgi.FieldStorage()
+
+    if 'pdf' not in form or not getattr(form['pdf'], 'file', None):
+        http_error(400, 'Missing PDF file.')
+    pdf_item = form['pdf']
+    footer_text = form.getfirst('footer_text', '')[:10000]  # cap length
```

Comment:
`footer_text` is never checked or sanitized, meaning that the corresponding argument to `add_footer.py` can containing arbitrary bytes/Unicode sequences. I wonder if funky input here could cause problems, but haven't checked this thoroughly.